src/router/index.js

```diff
@@ -7,24 +7,41 @@ const routes = [
   {
     path: "/",
     name: "Home",
-    component: Home,
+    component: Home
   },
   {
     path: "/play/:id",
     name: "Player",
     component: Player,
-    query: {src: ''}
+    query: {
+      src: ''
+    }
   },
   {
     path: "/login/:id?",
     name: "Phone Sign In",
     component: PhoneSignIn,
   },
-  { path: "/login", 
+  { 
+    path: "/login", 
     redirect: {
       name: "Phone Sign In" 
     } 
   },
+  {
+    path: '/404-not-found',
+    name: '404',
+    component: () => import("../views/Error"),
+    props: { type: '404' }
+  },
+  {
+    // Refer to: https://stackoverflow.com/a/64186073/7870587
+    path: '/:pathMatch(.*)*',
+    redirect: {
+      'name': '404'
+    }
+  }
+
```

Comment:
Whenever a random link is called, it redirects to the 404 page. Vue3 has weird syntax for default route.